src/lib/provable/test/dynamic-array.unit-test.ts

```diff
@@ -0,0 +1,312 @@
+import { UInt8, Field, ZkProgram } from 'o1js';
+import { DynamicArray } from '../dynamic-array.js';
+import { assert } from '../gadgets/common.js';
+
+function expectThrows(fn: () => void | Promise<void>, msg: string) {
+  let threw = false;
+  try {
+    fn();
+  } catch {
+    threw = true;
+  }
+  assert(threw, msg);
+}
+
+let List = ZkProgram({
+  name: 'dynamicarrays',
+  methods: {
+    pushAndPop: {
+      privateInputs: [],
+      async method() {
+        // Define classes of dynamic arrays for specific provable types
+        class Bytestring extends DynamicArray(UInt8, { capacity: 8 }) {}
+
+        // Initialize an empty dynamic array
+        let bytes = new Bytestring();
```

Comment:
I tried passing the `bytes` object into the program as a private input but got an error, something wrong with how we initiate the object internally - I think we should fix this since the user would arguably expect to pass in DynamicArrays as private inputs as well

Reply:
Solved with the updated constructor call inside `provable()`, thanks for pointing out!